src/app/components/pickers/dateRange.tsx

```diff
@@ -1,66 +1,135 @@
 import React, {memo} from 'react';
-import {StyleSheet, View, ViewStyle} from 'react-native';
+import {StyleSheet, View} from 'react-native';
+import Modal from 'react-native-modal';
 
+import Typography from '../typography';
 import DatePicker from './datePicker';
-import Linear from '../seperator/linear';
+import Button from '../button';
 
 import strings from '../../constant/strings';
+import colors from '../../constant/colors';
+import fonts from '../../constant/fonts';
 
 type Props = {
-  style?: ViewStyle;
-  startDate: Date | undefined;
-  endDate: Date | undefined;
-  onChange: (date: Date | undefined, isStart: boolean) => void;
+  isVisible: boolean;
+  toggleModal: () => void;
+  onSubmit: () => void;
+  startDate?: Date;
+  endDate?: Date;
+  onChangeStart: (date?: Date) => void;
+  onChangeEnd: (date?: Date) => void;
```

Comment:
If we are using `onSubmit`, so I think we don't require `onChangeStart` & `onChangeEnd`. We can manage the state for startDate and endDate in this component and pass when onSubmit called